front_end_pocket_poll/lib/imports/groups_manager.dart

```diff
@@ -424,4 +426,77 @@ class GroupsManager {
         .toUpperCase()
         .compareTo(a.getGroupName().toUpperCase()));
   }
+
+  static Future<ResultStatus<List<Category>>> getAllCategoriesList(
+      final String groupId) async {
+    ResultStatus<List<Category>> retVal = new ResultStatus(success: false);
+
+    Map<String, dynamic> jsonRequestBody = getEmptyApiRequest();
+    jsonRequestBody[RequestFields.ACTION] = CategoriesManager.getAction;
+    jsonRequestBody[RequestFields.PAYLOAD].putIfAbsent(GROUP_ID, () => groupId);
+
+    ResultStatus<String> response =
+        await makeApiRequest(apiEndpoint, jsonRequestBody);
+
+    if (response.success) {
+      try {
+        Map<String, dynamic> body = jsonDecode(response.data);
+        ResponseItem responseItem = new ResponseItem.fromJson(body);
+
+        if (responseItem.success) {
+          List<dynamic> responseJson = json.decode(responseItem.resultMessage);
+          retVal.data =
+              responseJson.map((m) => new Category.fromJson(m)).toList();
+          retVal.success = true;
+        } else {
+          retVal.errorMessage = "Unable to load categories.";
+        }
+      } catch (e) {
+        retVal.errorMessage = "Unable to load categories.";
+      }
+    } else if (response.networkError) {
+      retVal.errorMessage =
+          "Network error. Unable to load categories. Check internet connection.";
+    } else {
+      retVal.errorMessage = "Unable to load categories.";
+    }
+    return retVal;
+  }
+
+  static Future<ResultStatus<Category>> getEventCategory(
+      final String groupId, final String eventId) async {
+    ResultStatus<Category> retVal = new ResultStatus(success: false);
+
+    Map<String, dynamic> jsonRequestBody = getEmptyApiRequest();
+    jsonRequestBody[RequestFields.ACTION] = CategoriesManager.getAction;
+    jsonRequestBody[RequestFields.PAYLOAD].putIfAbsent(GROUP_ID, () => groupId);
+    jsonRequestBody[RequestFields.PAYLOAD]
+        .putIfAbsent(RequestFields.EVENT_ID, () => eventId);
+
+    ResultStatus<String> response =
+        await makeApiRequest(apiEndpoint, jsonRequestBody);
+
+    if (response.success) {
+      try {
+        Map<String, dynamic> body = jsonDecode(response.data);
+        ResponseItem responseItem = new ResponseItem.fromJson(body);
+
+        if (responseItem.success) {
+          List<dynamic> responseJson = json.decode(responseItem.resultMessage);
+          retVal.data = new Category.fromJson(responseJson[0]);

```

Comment:
I don't necessarily think using 0 to get the first element of a list is a magic number since that number won't reasonably change so long as we're always responding with a singleton list. In any case it is still a hard coded number which should probably be avoided just because, so I got rid of it by using the .first() method that's part of the List class in dart.